drivers/eeprom/eeprom_emulator.c

```diff
@@ -0,0 +1,751 @@
+/*
+ * Copyright (c) 2020 Laczen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * This driver emulates an EEPROM device in flash.
+ *
+ * The emulation represents the EEPROM in flash as a region that is a direct
+ * map of the EEPROM data followed by a region where changes to the EEPROM
+ * data is stored. Changes are written as address-data combinations. The size
+ * of such a combination is determined by the flash write block size and
+ * the size of the EEPROM (required address space), with a minimum of 4 byte.
+ * The EEPROM page needs to be a multiple of the flash page. Multiple EEPROM
+ * pages are also supported and increases the number of writes that can be
+ * performed.
```

Comment:
would be nice to have here a sentence which describes how EEPROM pages usage are rotated.